webapp/app/js/models/job.js

```diff
@@ -10,9 +10,11 @@ treeherder.factory('ThJobModel', [
     // ThJobModel is the js counterpart of job
 
     var ThJobModel = function(data) {
-        // creates a new instance of ThJobModel
-        // using the provided properties
-        angular.extend(this, data);
+        if (typeof data === "object" && data !== null) {
```

Comment:
We need a comment here to explain why we're not using angular.extend, otherwise someone might be tempted to change it back (since the original form is more concise).